riscv/src/register/mstatus.rs

```diff
@@ -477,44 +366,43 @@ impl Mstatus {
     /// In RISCV-32, this field is read from the [`crate::register::mstatush`] register
     pub fn mbe(&self) -> Endianness {
         match () {
-            #[cfg(riscv32)]
+            #[cfg(not(target_arch = "riscv32"))]
+            () => Endianness::try_from(bf_extract(self.bits, 37, 1)).unwrap_or_default(),
+            #[cfg(target_arch = "riscv32")]
             () => super::mstatush::read().mbe(),
-            #[cfg(not(riscv32))]
-            () => Endianness::from(bf_extract(self.bits, 37, 1) != 0),
         }
     }
+
     /// Update M-mode non-instruction-fetch memory endianness
     ///
     /// Note this updates a previously read [`Mstatus`] value, but does not
     /// affect the mstatus CSR itself. See [`set_mbe`] to directly update the
     /// CSR.
     #[inline]
```

Comment:
```suggestion
    ///
    /// **NOTE**: panics on RISCV-32 platforms.
    #[inline]
```

Reply:
Maybe in this kind of methods, for RV32, we should redirect users to use `mstatush`

Reply:
> we should redirect users to use `mstatush`

I agree, that would be helpful. I'll add it for `sbe` and `mbe`. These are the only two fields with overlap, right?